modules/cloud-config-container/simple-nva/files/policy_based_routing.sh

```diff
@@ -15,20 +15,27 @@
 # limitations under the License.
 
 IF_NAME=$1
-IP_LB=$(ip r show table local | grep "$IF_NAME proto 66" | cut -f 2 -d " ")
-
-# Sleep while there's no load balancer IP route for this IF
-while [ -z $IP_LB ] ; do
-  sleep 2
-  IP_LB=$(ip r show table local | grep "$IF_NAME proto 66" | cut -f 2 -d " ")
-done
-
 IF_NUMBER=$(echo $IF_NAME | sed -e s/eth//)
 IF_GW=$(curl http://metadata.google.internal/computeMetadata/v1/instance/network-interfaces/$IF_NUMBER/gateway -H "Metadata-Flavor: Google")
 IF_IP=$(curl http://metadata.google.internal/computeMetadata/v1/instance/network-interfaces/$IF_NUMBER/ip -H "Metadata-Flavor: Google")
 IF_NETMASK=$(curl http://metadata.google.internal/computeMetadata/v1/instance/network-interfaces/$IF_NUMBER/subnetmask -H "Metadata-Flavor: Google")
 IF_IP_PREFIX=$(/var/run/nva/ipprefix_by_netmask.sh $IF_NETMASK)
-grep -qxF "$((200 + $IF_NUMBER)) hc-$IF_NAME" /etc/iproute2/rt_tables || echo "$((200 + $IF_NUMBER)) hc-$IF_NAME" >>/etc/iproute2/rt_tables
-ip route add $IF_GW src $IF_IP dev $IF_NAME table hc-$IF_NAME
-ip route add default via $IF_GW dev $IF_NAME table hc-$IF_NAME
-ip rule add from $IP_LB/32 table hc-$IF_NAME
+
+# Sleep while there's no load balancer IP route for this IF
+while true
+do
+  IPS_LB_STR=$(ip r show table local | grep "$IF_NAME proto 66" | cut -f 2 -d " " |  tr -s '\n' ' ')
+  IPS_LB=($IPS_LB_STR)
+  for IP in "${IPS_LB[@]}"
+  do
+    # Configure hc routing table if not available for this network interface
+    grep -qxF "$((200 + $IF_NUMBER)) hc-$IF_NAME" /etc/iproute2/rt_tables || {
+      echo "$((200 + $IF_NUMBER)) hc-$IF_NAME" >>/etc/iproute2/rt_tables
+      ip route add $IF_GW src $IF_IP dev $IF_NAME table hc-$IF_NAME
+      ip route add default via $IF_GW dev $IF_NAME table hc-$IF_NAME
+    }
+    # configure PBR route for LB
+    ip rule list | grep -qF "$IP" || ip rule add from $IP/32 table hc-$IF_NAME
```

Comment:
can we please add something it also removes the route if the LB is no longer there?

Reply:
Done!